sympy/solvers/solveset.py

```diff
@@ -188,8 +187,13 @@ def _invert_real(f, g_ys, symbol):
                 return _invert_real(base, res, symbol)
 
         if not base_has_sym:
-            return _invert_real(expo,
-                imageset(Lambda(n, log(n)/log(base)), g_ys), symbol)
+            if base is not S.Zero:
+                return _invert_real(expo,
+                    imageset(Lambda(n, log(n)/log(base)), g_ys), symbol)
+            if base is S.Zero and g_ys.args[0] is S.One:
```

Comment:
> I'm not sure what g_ys is

`g_ys` is actually a `FiniteSet` that is returned by every recursive call.

> Would it ever have args[0] be 1 but other terms also present?

I don't think there will be more than one args in `g_ys` because it is a sub-condition of `if f.is_Pow` which will always have  single element in finiteset.

> 1 in position other than 0?

I guess this is a very special case `1` will be always present in position 0